pkg/resources/grant_privileges_to_share_acceptance_test.go

```diff
@@ -521,6 +521,45 @@ func TestAcc_GrantPrivilegesToShare_NoOnOption(t *testing.T) {
 	})
 }
 
+// proves https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2621 doesn't apply to this resource
+func TestAcc_GrantPrivilegesToShare_RemoveShareOutsideTerraform(t *testing.T) {
+	databaseName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	shareName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	configVariables := config.Variables{
+		"to_share": config.StringVariable(shareName),
+		"database": config.StringVariable(databaseName),
+		"privileges": config.ListVariable(
+			config.StringVariable(sdk.ObjectPrivilegeUsage.String()),
+		),
+	}
+
+	var shareCleanup func()
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					shareCleanup = createShareOutsideTerraform(t, shareName)
+				},
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_GrantPrivilegesToShare/OnCustomShare"),
+				ConfigVariables: configVariables,
+			},
+			{
+				PreConfig:       func() { shareCleanup() },
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_GrantPrivilegesToShare/OnCustomShare"),
+				ConfigVariables: configVariables,
+				// The error occurs in the Create operation, indicating the Read operation removed the resource from the state in the previous step.
+				ExpectError: regexp.MustCompile("An error occurred when granting privileges to share"),
```

Comment:
hmm, haven't we discussed that this could work like e.g. schema resource (check before granting, setID "" if object does not exist)? Or maybe it was other topic?

Reply:
Hmm, I'm pretty certain we wanted to do it in this way, where the check is in the Read operation and not the Create operation (schema also does that). I'm not sure if such a check is necessary for the Create, because it will fail anyway and won't create anything. Read matters for grant resources because without an additional check there, we will create a plan and fail at the Update operation which creates an infinite plan.

Reply:
Followed-up privately. Discussions resulted in: please add similar tests for schema to confirm that the behavior is the same.